lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -141,6 +141,17 @@ static bool isDuplicatableExpression(Operation *op) {
   if (auto array = dyn_cast<hw::ArrayGetOp>(op))
     return array.index().getDefiningOp<ConstantOp>();
 
+  // Inline simple CSE'd expressions
+  if (isa<AndOp, OrOp, XorOp>(op)) {
```

Comment:
We have some constraints which can't be solved here easily.  always event lists must not be expressions (often).  When they are not expressions, if inside them must use the same wire and not inline the expression.

The solution we are moving to is for export verilog to always inline and these kinds of choices are made in prepare.